src/coreclr/nativeaot/Runtime/unix/unixasmmacrosloongarch64.inc

```diff
@@ -0,0 +1,328 @@
+.endm
+
+.macro PREPARE_EXTERNAL_VAR_INDIRECT Name, HelperReg
+        la.local  \HelperReg, \Name
+        ld.d  \HelperReg, \HelperReg, 0
+.endm
+
+.macro PREPARE_EXTERNAL_VAR_INDIRECT_W Name, HelperReg
+        la.local  \HelperReg, \Name
+        ld.w  \HelperReg, \HelperReg, 0
+.endm
+
+
+.macro PROLOG_STACK_ALLOC Size
+        addi.d  $sp, $sp, -\Size
+.endm
+
+.macro EPILOG_STACK_FREE Size
+        addi.d  $sp, $sp, \Size
+        .cfi_adjust_cfa_offset -\Size
+.endm
+
+.macro EPILOG_STACK_RESTORE
+        ori  $sp, $fp, 0
+        .cfi_restore  3
+.endm
+
+.macro PROLOG_SAVE_REG reg, ofs
+        st.d  $r\reg, $sp, \ofs
+        .cfi_rel_offset \reg, \ofs
+.endm
+
+.macro PROLOG_SAVE_REG_PAIR reg1, reg2, ofs
+        st.d  $r\reg1, $sp, \ofs
+        st.d  $r\reg2, $sp, \ofs + 8
+        .cfi_rel_offset  \reg1, \ofs
+        .cfi_rel_offset  \reg2, \ofs + 8
+        .ifc  \reg1, $fp
+        ori  $fp, $sp, 0
+        .cfi_def_cfa_register  22
+        .endif
+.endm
+
+.macro PROLOG_SAVE_REG_PAIR_INDEXED reg1, reg2, ofs
+        st.d  $r\reg1, $sp, \ofs
+        st.d  $r\reg2, $sp, \ofs + 8
+        addi.d  $sp, $sp, \ofs
+        .cfi_adjust_cfa_offset  -\ofs
+        .cfi_rel_offset  \reg1, 0
+        .cfi_rel_offset  \reg2, 8
+        .ifc  \reg1, $fp
+        ori  $fp, $sp, 0
+        .cfi_def_cfa_register  $fp
+        .endif
+.endm
+
+.macro PROLOG_SAVE_REG_PAIR_NO_FP_INDEXED reg1, reg2, ofs
+        st.d  $r\reg1, $sp, \ofs
+        st.d  $r\reg2, $sp, \ofs + 8
+        addi.d  $sp, $sp, \ofs
+        .cfi_adjust_cfa_offset  -\ofs
+        .cfi_rel_offset  \reg1, 0
+        .cfi_rel_offset  \reg2, 8
+.endm
+
+
+.macro EPILOG_RESTORE_REG reg, ofs
+        ld.d  $r\reg, $sp, \ofs
+        .cfi_restore  \reg
+.endm
+
+.macro EPILOG_RESTORE_REG_PAIR reg1, reg2, ofs
+        ld.d  $r\reg1, $sp, \ofs
+        ld.d  $r\reg2, $sp, \ofs + 8
+        .cfi_restore  \reg1
+        .cfi_restore  \reg2
+.endm
+
+.macro EPILOG_RESTORE_REG_PAIR_INDEXED reg1, reg2, ofs
+        ld.d  $r\reg1, $sp, 0
+        ld.d  $r\reg2, $sp, 8
+        addi.d  $sp, $sp, \ofs
+        .cfi_restore  \reg1
+        .cfi_restore  \reg2
+        .cfi_adjust_cfa_offset  -\ofs
+.endm
+
+.macro EPILOG_RETURN
+        jirl  $r0, $ra, 0
+.endm
+
+.macro EMIT_BREAKPOINT
+        break  0
+.endm
+
+.macro EPILOG_BRANCH_REG reg
+
+        jirl  $r0, \reg, 0
+
+.endm
+
+// Loads the address of a thread-local variable into the target register,
+// which cannot be a0. Preserves all other registers.
+.macro INLINE_GET_TLS_VAR target, var
+    .ifc  \target, $a0
+        .error "target cannot be a0"
+    .endif
+
+    st.d  $a0, $sp, -0x10
+    st.d  $ra, $sp, -0x8
+    addi.d  $sp, $sp, -16
+
+    // This sequence of instructions is recognized and potentially patched
+    // by the linker (GD->IE/LE relaxation).
+    //la.local  $a0, \var                      //
+    //ld.d  \target, $a0, 0                    //
+    //.tlsdesccall \var                        //TODO-LOONGARCH64
```

Comment:
Anything preventing fixing this TODO?

Reply:
`TLSDESC` is a new feature on LoongArch that is not yet supported in LLVM, so that `.tlsdesccall` is not defined yet.
LLVM is expected to support TLS in a few months, and we need to wait for LLVM support before fixing it.
Do I need to delete this TODO?
Thanks!

Reply:
It is fine to keep the TODO, but it should explain when it is about. E.g. `TODO-LOONGARCH64: Fix once TLSDESC is supported by LLVM (add link to LLVM tracking issue if there is one)`

Reply:
@sunlijun-610  maybe the LLVM's had merged the LA64's TLS patch.

Reply:
OK, I will add the note of TODO first.
And I will ask the leader of LoongArch LLVM project. If the LA64's TLS patch has been already merged, I will fix it as soon as possible.
Thanks!